jdcoefct.c

```diff
@@ -381,21 +384,26 @@ smoothing_ok(j_decompress_ptr cinfo)
     /* All components' quantization values must already be latched. */
     if ((qtable = compptr->quant_table) == NULL)
       return FALSE;
-    /* Verify DC & first 5 AC quantizers are nonzero to avoid zero-divide. */
+    /* Verify DC & first 9 AC quantizers are nonzero to avoid zero-divide. */
     if (qtable->quantval[0] == 0 ||
         qtable->quantval[Q01_POS] == 0 ||
         qtable->quantval[Q10_POS] == 0 ||
         qtable->quantval[Q20_POS] == 0 ||
         qtable->quantval[Q11_POS] == 0 ||
-        qtable->quantval[Q02_POS] == 0)
+        qtable->quantval[Q02_POS] == 0 ||
+        qtable->quantval[Q03_POS] == 0 ||
+        qtable->quantval[Q12_POS] == 0 ||
+        qtable->quantval[Q21_POS] == 0 ||
+        qtable->quantval[Q30_POS] == 0)
       return FALSE;
     /* DC values must be at least partly known for all components. */
     coef_bits = cinfo->coef_bits[ci];
     prev_coef_bits = cinfo->coef_bits[ci + cinfo->num_components];
     if (coef_bits[0] < 0)
       return FALSE;
+    coef_bits_latch[0] = coef_bits[0];
```

Comment:
Is this necessary?  I can't see where `coef_bits[0]` is ever used.

Reply:
No, this is indeed not used, but in in line 55 or jdcoefct.h we had
`#define SAVED_COEFS  6          /* we save coef_bits[0..5] */`
and now this is
`#define SAVED_COEFS  10         /* we save coef_bits[0..9] */`
From that one might expect that coef_bits[0] is also stored. Otherwise one could make the size of SAVED_COEFS one smaller I guess. However I propse to keep the size.